src/main/java/in/nimbo/dao/elastic/ElasticDAOImpl.java

```
@@ -0,0 +1,94 @@
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class ElasticDAOImpl implements ElasticDAO {
+    private final ElasticConfig config;
+    private RestHighLevelClient client;
+
+    public ElasticDAOImpl(ElasticConfig config, RestHighLevelClient client) {
+        this.config = config;
+        this.client = client;
+    }
+
+    public ElasticDAOImpl(ElasticConfig config) {
+        this.config = config;
+        client = new RestHighLevelClient(RestClient.builder(new HttpHost(config.getHost(), config.getPort())));
+    }
+
+    @Override
+    public void save(String link, String text) {
+        try {
+            IndexRequest request = new IndexRequest(config.getIndexName()).id(link).type(config.getType());
+
+            XContentBuilder builder = XContentFactory.jsonBuilder();
+            builder.startObject();
+            builder.field("text", text);
+            builder.endObject();
+            request.source(builder);
+            IndexResponse index = client.index(request, RequestOptions.DEFAULT);
+            if (index.getResult() != DocWriteResponse.Result.CREATED)
+                throw new ElasticException("Indexing failed!");
+        } catch (IOException e) {
+            throw new ElasticException("Indexing failed!", e);
+        }
+    }
+
+    @Override
+    public Optional<String> get(String link) {
+        try {
+            GetRequest getRequest = new GetRequest(config.getIndexName(), config.getType(), link);
+            GetResponse response = client.get(getRequest, RequestOptions.DEFAULT);
+            if (response.isExists()) {
+                String text = (String) response.getSource().get("text");
+                return Optional.ofNullable(text);
+            }
+        } catch (IOException e) {
+            throw new ElasticException("Get failed", e);
+        } catch (ClassCastException ignored) {
```

Do not use empty catch block.
Use optional.nullable instead